healsparse/healSparseMap.py

```diff
@@ -1294,11 +1318,24 @@ def degrade(self, nside_out, reduction='mean'):
                 aux_dtype = np.float64
             else:
                 aux_dtype = self._sparse_map.dtype
+            if weights is not None:
+                if reduction == 'wavg':
+                    bad_map = ((weights.nside_sparse != self.nside_sparse) or
+                               (weights.nside_coverage != self.nside_coverage) or
+                               (not np.allclose(weights.valid_pixels, self.valid_pixels)))
```

Comment:
Why is this guard not the same as above (`weights.valid_pixels == self.valid_pixels`)?

Reply:
Oh, that's a bug! `weights.valid_pixels != self.valid_pixels` doesn't work (it throws a `ValueError` because I'm comparing a bitwise quantity with an ndarray of bools. I just put `weights.valid_pixels.all() != self.valid_pixels.all()`

Reply:
That doesn't make sense ... `valid_pixels` should be an array of integers.  How is it not?

Reply:
Oh, the problem is that you need `np.array_equal(weights.valid_pixels, self.valid_pixels)`.

Reply:
(The `weights.valid_pixels.all() != self.valid_pixels.all()` doesn't work either, by the way.).